### PR TITLE
Poppy flowers react less often during rain.

### PR DESCRIPTION
**Pull request by RustyPotato**
As following with int's suggestion on the Discord mod-and-development board, 2/18/18 at 0941. 

"What if we change special 'attack' of poppy flower at iexamine::flower_poppy() to be less common at rainy weather? Like twice rarer?"

I decided I could do that. So I did.

### CODE REVIEW

File: src/iexamine.cpp [hunk 1]
```diff
@@ -1444,7 +1444,11 @@ void iexamine::flower_poppy(player &p, const tripoint &examp)
         add_msg(m_warning, _("This flower has a heady aroma."));
     }
 
-    if (one_in(3) && resist < 5)  {
+    auto recentWeather = sum_conditions( calendar::turn.get_turn()-10_minutes, calendar::turn.get_turn(), p.pos() );
```
Comment (BevapDin) [inline src/iexamine.cpp]: You don't need the `.get_turn()`, you can use `calendar::turn` directly.

**Comment by kevingranade** [comment]
Logic error, this makes it 50% more likely.

**Comment by illi-kun** [comment]
@RustyPotato 
The only blocker of this PR is a broken logic of conditional expression described by Kevin. Fix this and we're good to go.

**Comment by RustyPotato** [comment]
Sorry for the delay everyone. I've fixed it according to the suggestions.

**Pull request merged**